File: CSharp/PaymentsService/Domain/Payments/ProcessPaymentBaseResponse.cs
namespace PaymentsService.Domain.Payments;

public abstract class ProcessPaymentBaseResponse
{
    public ProcessPaymentBaseResponse(string paymentMethod)
    {
        PaymentMethod = paymentMethod;
    }

    public string PaymentMethod { get; set; }
    public string Status { get; set; }
    public string Error { get; set; }
}

File: Python/PaymentsService/Domain/Payments/ProcessPaymentBaseResponse.py
from abc import ABC, abstractmethod

class ProcessPaymentBaseResponse(ABC):
    def __init__(self, payment_method, status = None, error = None):
        self.paymentMethod = payment_method
        self.status = status
        self.error = error
    
    @abstractmethod
    def to_dict(self) -> dict:
        pass